src/Strategy/CommonClassesStrategy.php

```diff
@@ -68,13 +69,13 @@ final class CommonClassesStrategy implements DiscoveryStrategy
             ['class' => SlimUriFactory::class, 'condition' => [SlimRequest::class, SlimUriFactory::class]],
         ],
         HttpAsyncClient::class => [
-            ['class' => SymfonyHttplug::class, 'condition' => [SymfonyHttplug::class, Promise::class, RequestFactory::class]],
+            ['class' => SymfonyHttplug::class, 'condition' => [SymfonyHttplug::class, Promise::class, RequestFactory::class, [CommonClassesStrategy::class, 'isPsr17FactoryInstalled']]],
```

Comment:
How about `self::class` instead of `CommonClassesStrategy::class`?

Reply:
What is the difference?

Reply:
It looks like the style guide agrees with you =)